_lessons/python-scatter/06-profiling_with_MAP.md

```diff
@@ -1,77 +1,79 @@
 ---
 layout: post
-title: Profiling_MAP
+title: Profiling with MAP
 permalink: /python-scatter/profiling_MAP
 chapter: python-scatter
 ---
 
 ## Objectives
 
-Learn how to profile parallel codes using ARM MAP:
+You will:
 
-* learn how to use MAP
-* visualise the profiling data
-* interpret the profiling results
+* learn how to use MAP to profile an MPI code
+* learn how to interpret MAP profiling data
 
 ## Introduction to profiling parallel codes
 
 Parallel profiling tools provide information about how much time is spent in different parts of your code by different threads/processes.
-There are 2 major ways to collect profiling data: *sampling* and *tracing*. Sampling is a statistical method. During run time the so called *stack trace* is analysed regularly. In contast to that tracing measures actual run time of instrumented parts with a well defined start and end. The data from all threads/processes are gathered and analysed. Thus we can not only see which part of the code is most time consuming, but also how this performs on different processes.
+There are two major ways to collect profiling data: 
+ * *sampling*: data are statistically harvested during run time to determine which parts of the code are most time consuming. This is useful for identiying the hotspots in a code (see [profiling](https://nesi.github.io/perf-training/python-scatter/profiling)).
+ * *tracing*: records the activity of instrumented parts of the code in time. This is useful to determine what each thread and process is doing at a particaular point in time. Tracing can reveal load balancing issues or the communication patterns of an MPI program. 
 
-Due to possible overhead (especially for tracing experiments) the program could run significantly slower than normal. Therefore it is generally advisable to first run a sampling experiment and choose a small representative test case to profile.
+MAP can be used both for sampling and tracing.
 
-## Profiling test case
+## Choosing your test case
 
-The profiling case should be:
-* representative, need to cover all features used during production
-* short in run time for a fast turnaround, but long enough to be representative of the actual calculations you are going to run
-* scaling effects can be investigated by comparing profiles small and big jobs
+Choose your test case carefully, making sure:
+* it runs in a short time. Due to a possible large overhead from MAP (especially for tracing experiments), the program could run significantly slower than normal.
+* ensure the test case is representative of a production run
```

Comment:
Do not miss that the run time should not be too short esp. for sampling. Otherwise, we end up with just a few samples and these will not lead to representative analysis.